### PR TITLE
Allow fullscreen (Mac OS)

### PR DESCRIPTION
**Pull request by BoogieMonster1O1**
Added support for fullscreen (works only on Mac OS)

Before: 
Screen Shot 2020-06-30 at 7 05 20 PM

After: 
Screen Shot 2020-06-30 at 7 04 56 PM

### CODE REVIEW

File: enigma-swing/src/main/java/cuchaz/enigma/gui/Gui.java [hunk 1]
```diff
@@ -363,6 +367,14 @@ public void windowClosing(WindowEvent event) {
...
 		this.frame.setDefaultCloseOperation(WindowConstants.DO_NOTHING_ON_CLOSE);
 		this.frame.setLocationRelativeTo(null);
+		
+		try {
+            Class<?> clazz = Class.forName("com.apple.eawt.FullScreenUtilities");
+            Method method = clazz.getMethod("setWindowCanFullScreen", new Class<?>[] {
+                    Window.class, boolean.class });
+            method.invoke(null, frame, true);
+        } catch (Throwable ignored) {
+        }
```
Comment (Juuxel) [inline enigma-swing/src/main/java/cuchaz/enigma/gui/Gui.java:377]: Uh, what is this indentation? :concern:

**Comment by Earthcomputer** [comment]
Why not use the normal method of making a swing application fullscreen, undecorated frames?

File: enigma-swing/src/main/java/cuchaz/enigma/gui/Gui.java (hunk 1 above)
Comment (BoogieMonster1O1) [reply enigma-swing/src/main/java/cuchaz/enigma/gui/Gui.java]: fixed

**Comment by BoogieMonster1O1** [comment]
> Why not use the normal method of making a swing application fullscreen, undecorated frames? That's cross-platform too.

I tried using the `GraphicsDevice` method but it didn't work properly

**Comment by BoogieMonster1O1** [comment]
> Why not use the normal method of making a swing application fullscreen, undecorated frames? That's cross-platform too.

also, setting the frame as undecorated doesn't affect fullscreen at all

**Comment by BoogieMonster1O1** [comment]
There's a better solution to this, so I'm closing this now

**Pull request closed without merging**